coral-common/src/main/java/com/linkedin/coral/common/HiveTypeSystem.java

```diff
@@ -161,6 +161,15 @@ public boolean isSchemaCaseSensitive() {
     return false;
   }
 
+  /**
+   * This method needs to be overridden to make sure that the "/" operator returns
+   * {@link org.apache.calcite.sql.type.ReturnTypes#DOUBLE}, which is compatible with the expected data type in Hive/Spark.
+   */
+  @Override
+  public RelDataType deriveDecimalDivideType(RelDataTypeFactory typeFactory, RelDataType type1, RelDataType type2) {
+    return nullableType(typeFactory, SqlTypeName.DOUBLE);
```

Comment:
super's implementation here https://github.com/apache/calcite/blob/main/core/src/main/java/org/apache/calcite/rel/type/RelDataTypeSystem.java#L277 offers some additional value in tracking precision & scale, when either of the operands is decimal type. That could be helpful information for Coral consumers & will remove redundant CASE statements in the affected translated SQLs.  

Can we override the behavior only for use cases when both operands are integers and rely on the super's implementation for other scenarios?

Reply:
Good point, actually all the other cases (like bigint divide int) return `DOUBLE` except for the case in super's implementation. Modified the implementation.